From bfc14d067da0fdf65f306fcd9a65829d9694dbeb Mon Sep 17 00:00:00 2001
From: John Keeping <john@keeping.me.uk>
Date: Sat, 02 Mar 2013 12:32:10 +0000
Subject: Update git to v1.7.5.4

Some changes to diff options:

- no_merges has become the more general max_parents
- path restriction now uses struct pathspec

Signed-off-by: John Keeping <john@keeping.me.uk>
---
diff --git a/git b/git
-Subproject 7ed863a85a6ce2c4ac4476848310b8f917ab41f
+Subproject e5af0de202e885b793482d416b8ce9d50dd2b8b
diff --git a/shared.c b/shared.c
index 8e5ae48..0967f92 100644
--- a/shared.c
+++ b/shared.c
@@ -307,7 +307,7 @@ void cgit_diff_tree(const unsigned char *old_sha1,
 		    filepair_fn fn, const char *prefix, int ignorews)
 {
 	struct diff_options opt;
-	int prefixlen;
+	struct pathspec_item item;
 
 	diff_setup(&opt);
 	opt.output_format = DIFF_FORMAT_CALLBACK;
@@ -319,10 +319,10 @@ void cgit_diff_tree(const unsigned char *old_sha1,
 	opt.format_callback = cgit_diff_tree_cb;
 	opt.format_callback_data = fn;
 	if (prefix) {
-		opt.nr_paths = 1;
-		opt.paths = &prefix;
-		prefixlen = strlen(prefix);
-		opt.pathlens = &prefixlen;
+		item.match = prefix;
+		item.len = strlen(prefix);
+		opt.pathspec.nr = 1;
+		opt.pathspec.items = &item;
 	}
 	diff_setup_done(&opt);
 
diff --git a/ui-stats.c b/ui-stats.c
index 59f4c1e..51dd6a2 100644
--- a/ui-stats.c
+++ b/ui-stats.c
@@ -239,7 +239,7 @@ struct string_list collect_stats(struct cgit_context *ctx,
 	init_revisions(&rev, NULL);
 	rev.abbrev = DEFAULT_ABBREV;
 	rev.commit_format = CMIT_FMT_DEFAULT;
-	rev.no_merges = 1;
+	rev.max_parents = 1;
 	rev.verbose_header = 1;
 	rev.show_root_diff = 0;
 	setup_revisions(argc, argv, &rev, NULL);
--
cgit v0.9.1-37-gd50e
